### ISSUE TITLE
Side effect of tf.global_variables_initializer on evaluation of output model

### ISSUE THREAD

**Issue by ngonthier** [keras-team/keras#13335]
**System information**  
- Have I written custom code that you can find here :

`

import tensorflow as tf
from tensorflow.python.keras.preprocessing import image as kp_image
from tensorflow.python.keras import models 
import numpy as np

def minimalCase_forIssue():
    max_dim = 224
    path_to_img =  'Q23898.jpg' # Need to change with the name of your image
    img = tf.keras.preprocessing.image.load_img(
        path_to_img,
        grayscale=False,
        target_size=(max_dim, max_dim),
        interpolation='nearest')
    img = kp_image.img_to_array(img)
    img = np.expand_dims(img, axis=0) # Should be replace by expand_dims in tf
    img = tf.keras.applications.vgg19.preprocess_input(img)
    img_tensor = tf.convert_to_tensor(img)
    
    vgg = tf.keras.applications.vgg19.VGG19(include_top=False, weights='imagenet')
    vgg.trainable = False
    output = vgg.get_layer('block1_conv1').output
    model = models.Model(vgg.input, output)
    sess = tf.Session()
    
    for i in range(2):
        sess.run(tf.global_variables_initializer())
        evaluation_by_sess = sess.run(model(img_tensor))
        print('With sess.run :',evaluation_by_sess[0,0,0,:])
        evaluation_by_predict = model.predict(img,batch_size=1)
        print('With .predict() :',evaluation_by_predict[0,0,0,:])
    sess.close()
    
if __name__ == '__main__':
    minimalCase_forIssue()

`

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.8.0
- Keras version:  2.1.5
- Python version:  3.6

When I use sess.run to obtain the output of a model (here the first layer of a pretrained VGG19), I obtain a different output that the one obtained by using the predict function of the model. I am looking at an inference output. There is no training at all. All the weights are freeze and not trainable.

Moreover each time, I run tf.global_variables_initializer, I will obtain a different result whereas there is no stochastic element in the code.

You can find a minimal code above.
